viewing_party/party.py

```diff
@@ -1,23 +1,168 @@
+import copy
 # ------------- WAVE 1 --------------------
+movies = {}
 
 def create_movie(title, genre, rating):
-    pass
+    if title != None and genre != None and rating != None:
+        movies.update({"genre": genre, "rating": rating, "title": title})
+        return movies
+    
+    else:
+        return None
 
+def add_to_watched(user_data, movie):
+    user_data.update({"watched": [movie]})
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data.update({"watchlist": [movie]})
+    return user_data
+
+def watch_movie(user_data, title):
+
+    #work through the whole watchlist; copy it first so no DANGERZONE/index/key errors
+    copy_list = user_data["watchlist"].copy()
+    for i in range(len(copy_list)):
+
+    #find the title of the movie in the watchlist
+        if copy_list[i]["title"] == title:
+
+            #if the title is in the watchlist, move it to watched then delete
+            user_data["watched"].append(copy_list[i])
+            del user_data["watchlist"][i]
+
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    total = 0
+
+    #make sure the dictionary isn't empty
+    if user_data["watched"]:
+
+        #iterate over the watched list and add averages to total
+        for i in range(len(user_data["watched"])):
+            total += user_data["watched"][i]["rating"]

```

Comment:
We only use i to index into our array for the current item. Since we don't need i as a counter for anything else, we could simplify some of our syntax by looping directly over the elements of the list:
```suggestion
        for movie in user_data["watched"]:
            total += movie["rating"]
```